test/test_recog.py

```diff
@@ -0,0 +1,145 @@
+# coding: utf-8
+
+# Copyright 2018 Hiroshi Seki
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+import lm_pytorch
+
+import lm_chainer
+
+import argparse
+import importlib
+import numpy
+
+import pytest
+
+
+def make_arg(**kwargs):
+    defaults = dict(
+        elayers=4,
+        subsample="1_2_2_1_1",
+        etype="blstmp",
+        eunits=100,
+        eprojs=100,
+        dlayers=1,
+        dunits=300,
+        atype="location",
+        aconv_chans=10,
+        aconv_filts=100,
+        mtlalpha=0.5,
+        lsm_type="",
+        lsm_weight=0.0,
+        adim=320,
+        dropout_rate=0.0,
+        nbest=5,
+        beam_size=3,
+        penalty=0.5,
+        maxlenratio=1.0,
+        minlenratio=0.0,
+        ctc_weight=0.2,
+        verbose=2,
+        char_list=["a", "i", "u", "e", "o"],
+        outdir=None,
+        ctc_type="chainer"
+    )
+    defaults.update(kwargs)
+    return argparse.Namespace(**defaults)
+
+
+def init_torch_weight_const(m, val):
+    for p in m.parameters():
+        p.data.fill_(val)
+
+
+def init_chainer_weight_const(m, val):
+    for p in m.params():
+        p.data[:] = val
+
+
+@pytest.mark.parametrize(("etype", "m_str", "text_idx1"), [
+    ("blstmp", "e2e_asr_attctc", 0),
+    ("blstmp", "e2e_asr_attctc_th", 1),
+    ("vggblstmp", "e2e_asr_attctc", 2),
+    ("vggblstmp", "e2e_asr_attctc_th", 3),
+])
+def test_recognition_results(etype, m_str, text_idx1):
+    const = 1e-4
+    numpy.random.seed(1)
+    seq_true_texts = ([["o", "iuiuiuiuiuiuiuiuo", "iuiuiuiuiuiuiuiuo"],
+                       ["o", "uiuiuiuiuiuiuiuio", "uiuiuiuiuiuiuiuio"],
+                       ["o", "iuiuiuiuiuiuiuiuo", "iuiuiuiuiuiuiuiuo"],
+                       ["o", "uiuiuiuiuiuiuiuio", "uiuiuiuiuiuiuiuio"]])
+
+    for text_idx2, ctc_weight in enumerate([0.0, 0.5, 1.0]):
```

Comment:
Better to add comments that it's testing attention (0.0), hybrid CTC/attention (0.5), and CTC (1.0)